util/throttle/throttle.go

```diff
@@ -0,0 +1,36 @@
+package throttle
+
+import (
+	"sync"
+	"time"
+)
+
+func Throttle(d time.Duration, f func()) func() {
+	var next, running bool
+	var mu sync.Mutex
+	return func() {
+		mu.Lock()
+		defer mu.Unlock()
+
+		next = true
+		if !running {
+			running = true
```

Comment:
Can we use sync.Once?

Reply:
Can you explain more? `running` is not set only once. The goroutine can return and rerun again.